### ISSUE TITLE
[no-unsafe-argument] [REG 4->5] cannot pass bind as an argument

### ISSUE THREAD

**Issue by orgads** [typescript-eslint/typescript-eslint#4001]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-argument": "error"
  }
}
```

```TS
import { EventEmitter } from 'events';

export class Foo extends EventEmitter {
  constructor() {
    super();
    this.on('error', this.onError.bind(this));
  }
  onError(err: Error) { console.error(err); }
}
```

**Expected Result**

It should be valid.

**Actual Result**

`bind` returns `any`, so passing it to `on` yields an error:
```
Unsafe argument of type `any` assigned to a parameter of type `(...args: any[]) => void`.
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.0.0` |
| `@typescript-eslint/parser`        | `5.0.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `8.0.0` |
| `node`                             | `14.17.6` |
